frontend/src/registration/components/InstanceSetupPage/InstanceSetupPage.tsx

```diff
@@ -4,33 +4,66 @@ import { Form, FormControl, FormGroup, FormLabel } from 'react-bootstrap';
 import { connect } from 'react-redux';
 import { WrappedMessage } from 'utils/intl';
 import { DomainSuccessJumbotron } from 'ui/components';
-import { submitRegistration } from '../../actions';
+import { RegistrationNavButtons } from 'registration/components';
+import { ROUTES } from 'global/constants';
+import { validateInstanceInfo, updateInstanceInfoState } from '../../actions';
 import { getRegistrationData } from '../../selectors';
 import { RegistrationPage } from '../RegistrationPage';
 import messages from './displayMessages';
 import './styles.scss';
 
 interface ActionProps {
-  submitRegistration: typeof submitRegistration;
+  validateInstanceInfo: Function;
+  updateInstanceInfoState: Function;
 }
 
 interface StateProps {
   domain: string;
   domainIsExternal: boolean;
+  instanceName: string;
+  publicContactEmail: string;
+  loading: boolean;
 }
 
 interface Props extends StateProps, ActionProps {}
 
 @connect<StateProps, ActionProps, {}, Props, RootState>(
   (state: RootState) => ({
     domain: getRegistrationData(state, 'domain'),
-    domainIsExternal: getRegistrationData(state, 'domainIsExternal')
+    domainIsExternal: getRegistrationData(state, 'domainIsExternal'),
+    instanceName: getRegistrationData(state, 'instanceName'),
+    publicContactEmail: getRegistrationData(state, 'publicContactEmail'),
+    loading: getRegistrationData(state, 'loading')
   }),
   {
-    submitRegistration
+    validateInstanceInfo,
+    updateInstanceInfoState
   }
 )
-export class InstanceSetupPage extends React.PureComponent<Props> {
+export class InstanceSetupPage extends React.PureComponent<Props, StateProps> {
```

Comment:
The props of this page (and all other ones in the registration flow) will be lost after refresh. We could make it persistent, e.g. with [redux-storage](https://github.com/react-stack/redux-storage).

Reply:
I think this is the intended behavior of the current implementation. If the user want's to change anything, he can go back using the navigation buttons or browser back button.
If the page is refreshed it should start from scratch.

Reply:
![domain](https://user-images.githubusercontent.com/3189670/70260202-126bb700-1790-11ea-8c7f-013591395e93.gif)
I don't think it's a user-friendly behavior ;)

> If the page is refreshed it should start from scratch.

From the user perspective I would expect data to be retained for the completed steps when I refresh the page, especially as these steps have different routes.

Reply:
Oh, now I got what you meant. Yeah, this was actually a bug in the reducers that erased and replaced the global state. 
It's fixed now (I tested by going back and forth multiple times).

Reply:
Ok, this was partially fixed :stuck_out_tongue: 
I will add `redux-storage` to fix this.